File: Golang/136. Single Number.go
// func singleNumber(nums []int) int {
//     hashtable := make(map[int]int, len(nums)/2+1)
//     for _, num := range nums {
//         if hashtable[num] == 0 {
//             hashtable[num] = 1
//         } else {
//             delete(hashtable, num)
//         }
//     }
//     ans := 0
//     for k, _ := range hashtable {
//         ans = k
//     }
//     return ans
// }

func singleNumber(nums []int) int {
    ans := 0
    
    for _, num := range nums {
        ans ^= num // a^b^c^b^c = a
    }
    
    return ans
}

File: Python3/136. Single Number.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # hashtable = {}
        # for num in nums:
        #     if num in hashtable:
        #         del hashtable[num]
        #     else:
        #         hashtable[num] = True
        # return list(hashtable.keys())[0]
        
        ans = 0
        for num in nums:
            ans ^= num # a^b^c^b^c = a 这种方法简洁，但是速度慢
        return ans
